Detailed user information not requested if the users closes the app at registration

rishabhkohli opened the pull request.

andre-stefanov commented on app/src/main/java/app/nexd/android/ui/MainActivity.kt:
```diff
@@ -26,8 +26,13 @@ class MainActivity : AppCompatActivity() {
                     Preferences.getToken(this)?.let {
-                        Log.v("Navigation", "redirect to roleFragment")
-                        controller.navigate(R.id.roleFragment)
+                        if (Preferences.getRegistrationComplete(this)) {
+                            Log.v("Navigation", "redirect to roleFragment")
+                            controller.navigate(R.id.roleFragment)
+                        } else {
+                            Log.v("Navigation", "redirect to registerDetailedViewModel")
+                            controller.navigate(R.id.registerDetailedFragment)
+                        }
```
this logic should happen in a viewmodel

rishabhkohli replied:
Done.

andre-stefanov commented:
Please resolve merge conflicts

rishabhkohli commented:
Rebased the branch since the changes to develop were very substantial.
Also squashed the earlier 2 commits into one, because the earlier structure did not make any sense anymore.
Lastly, there was a bug on develop involving a crash on the RegisterDetailedFragment.
Fixed that in a separate commit on this branch.
